pkg/apis/kubebind/v1alpha1/apiservicebindingrequest_types.go

```diff
@@ -76,13 +82,16 @@ func (in *APIServiceBindingRequest) SetConditions(conditions conditionsapi.Condi
 type APIServiceBindingRequestSpec struct {
 	// parameters holds service provider specific parameters for this binding
 	// request.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="parameters are immutable"
```

Comment:
i believe this is optional? If so, we need a top-level CEL a la `// +kubebuilder:validation:XValidation:rule="!has(oldSelf. parameters) || has(self. parameters)", message="parameters are required once set"` to disallow removal, see https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification.

Reply:
this would be a CEL on the `APIServiceBindingRequestSpec` struct.

Reply:
oh fun. Thank for pointing that out.